### PR TITLE
RaR: Building Blocks

### PR DESCRIPTION
**Pull request by jwarwick**

### CODE REVIEW

**Comment by NoahTheDuke** [comment]
Might want to check that both costs are ignored.

File: src/clj/game/cards/operations.clj [hunk 1]
```diff
@@ -239,6 +239,14 @@
     :msg "do 7 meat damage"
     :effect (effect (damage eid :meat 7 {:card card}))}
 
+   "Building Blocks"
+   {:choices {:req #(and (= (:side %) "Corp")
+                         (ice? %)
```
Comment (Saintis) [inline src/clj/game/cards/operations.clj]: Technically the card only specified it has to be a Barrier, that it is also a piece of ICE is an assumption on our part.

File: test/clj/game_test/cards/operations.clj [hunk 2]
```diff
@@ -180,6 +180,20 @@
...
+    (new-game (default-corp ["Building Blocks" "Ice Wall"])
+              (default-runner))
+    (core/gain state :corp :credit 1)
+    (play-from-hand state :corp "Building Blocks")
+    (is (= 1 (:credit (get-corp))) "Corp starts with 1 credit")
+    (prompt-select :corp (find-card "Ice Wall" (:hand (get-corp))))
+    (prompt-choice :corp "New remote")
+    (let [iw (get-ice state :remote1 0)]
+      (is (= 1 (:credit (get-corp))) "Corp spent no credits")
+      (is (:rezzed (refresh iw)) "Ice Wall is installed and rezzed"))))
```
Comment (Saintis) [inline test/clj/game_test/cards/operations.clj]: I'd also like a test where you try to select a non-Barrier, and maybe a non-Barrier, non-ice as well.

File: test/clj/game_test/cards/operations.clj [hunk 3]
```diff
@@ -180,6 +180,20 @@
...
     (is (= 7 (count (:hand (get-corp)))) "Drew 2 cards")))
 
+(deftest building-blocks
+  ;; Building Blocks - install and rez a barrier from HQ at no cost
+  (do-game
+    (new-game (default-corp ["Building Blocks" "Ice Wall"])
+              (default-runner))
+    (core/gain state :corp :credit 1)
+    (play-from-hand state :corp "Building Blocks")
+    (is (= 1 (:credit (get-corp))) "Corp starts with 1 credit")
```
Comment (Saintis) [inline test/clj/game_test/cards/operations.clj]: I'd like this to check that Building Blocks cost 5 to play. Same effect, but different error message.

**Pull request merged**